packages/@aws-cdk/core/lib/stack.ts

```diff
@@ -753,6 +768,17 @@ export class Stack extends CoreConstruct implements ITaggable {
 
     // write the CloudFormation template as a JSON file
     const outPath = path.join(builder.outdir, this.templateFile);
+
+    if (this.node.tryGetContext(cxapi.VALIDATE_STACK_RESOURCE_LIMIT)) {
+      const resources = template.Resources || {};
+      const numberOfResources = Object.keys(resources).length;
+
+      if (numberOfResources > this.maxResources) {
+        throw new Error(`Number of resources: ${numberOfResources} is greater than allowed maximum of ${this.maxResources}`);
+      } else if (numberOfResources >= (this.maxResources * 0.8)) {
+        Annotations.of(this).addWarning(`Number of resources: ${numberOfResources} is approaching allowed maximum of ${this.maxResources}`);
```

Comment:
Please change this into `addMessage`.

A `warning` may have the effect of breaking deployment, if deployment is run using `--strict` (which turns warnings into errors).

This is a warning, but it's a different kind of warning from the regular ones.

Reply:
Would you mean `addInfo`?

Because the `addMessage` is a private method from `Annotations`!

Reply:
Oh, doh.